### ISSUE TITLE
Suggested way to load Matomo script only in production mode?

### ISSUE THREAD

**Issue by dosstx** [AmazingDreams/vue-matomo#23]
We are getting hits from development mode. What is the suggested method for loading the vue-matomo script for production builds only using Vue-CLI?

**Comment by dosstx** [comment]
@AmazingDreams I ended up keeping it simple and doing as you suggested:

`if (process.env.NODE_ENV === 'production') {
  Vue.use(VueMatomo, config)
}`

That seems to work....but I also had to do the same to some of my components that contained additional code such as:

```
created: function() {
    if (process.env.NODE_ENV == "production") {
      window._paq.push(["setCustomUrl", "/" + window.location.hash.substr(1)]);
      window._paq.push(["setDocumentTitle", "Overview Page"]);
      window._paq.push(["trackPageView"]);
    }
  }
```

If anyone has any additional information (or better ways to do this), let me know! Thanks.

**Comment by AmazingDreams** [comment]
You shouldn't need the second part of code (window._paq.push).

**Comment by dosstx** [comment]
Thanks. So I don't need to follow these steps with my Single page app?
https://developer.matomo.org/guides/spa-tracking

**Comment by dosstx** [comment]
Furthermore, I decided to add to a check for hostname:

```if (
  process.env.NODE_ENV == 'production' &&
  window.location.hostname == 'fgs-dps.gs.mil'
)
```

**Comment by AmazingDreams** [comment]
No vue-matomo takes care of the SPA. However your issue has brought to my attention that perhaps there's a need for a configuration flag to 'mock' the tracker and not actually send events. As your code might be using `$matomo` now while the value isn't set anywhere

**Comment by AmazingDreams** [comment]
Thanks! Glad it's of use to you!

**Comment by chrisl-peopleplus** [comment]
Hi @AmazingDreams Was there ever a PR/change implemented for the above comment to "mock" the tracker?

**Comment by AmazingDreams** [comment]
@chrisl-peopleplus no, but somewhat related is this commit / file https://github.com/AmazingDreams/vue-matomo/commit/c88d3f0cbdd357ce72c83af00241b0f5f9cf5ed1#diff-838cfd8d52586d01c57707bc42280394ce85aa28827322ad85d9c5355e269416

It does not work, because older browsers do not support Proxy. However it may be worthwhile to check it out again because Matomo and Vue itself may not support these browsers anymore either. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy#browser_compatibility

**Comment by chrisl-peopleplus** [comment]
If it helps count this as a +1 for the feature to be able not to send tracking data in a dev environment. At the moment I just wrapped the trackEvent in a `isProduction` test and told the people who use it for stats to ignore anything from localhost or UAT e.g. 

```
if (process.env.NODE_ENV === 'production') {
    this.$matomo.trackEvent;
}
```

Thanks for your work on this plugin, super useful!
